src/devices/PiJuice/PiJuicePower.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Iot.Device.PiJuiceDevice.Models;
+
+namespace Iot.Device.PiJuiceDevice
+{
+    /// <summary>
+    /// PiJuicePower class to support status of the PiJuice
+    /// </summary>
+    public class PiJuicePower
+    {
+        private readonly PiJuice _piJuice;
+
+        /// <summary>
+        /// PiJuicePower constructor
+        /// </summary>
+        /// <param name="piJuice">The PiJuice class</param>
+        public PiJuicePower(PiJuice piJuice)
+        {
+            _piJuice = piJuice;
+        }
+
+        /// <summary>
+        /// Gets the delay before the PiJuice removes power to the GPIO pins
+        /// </summary>
+        /// <returns>The delay in seconds</returns>
+        public byte GetPowerOff()
+        {
+            var response = _piJuice.ReadCommand(PiJuiceCommand.PowerOff, 1);
+
+            return response[0];
+        }
+
+        /// <summary>
+        /// Sets the delay before the PiJuice removes power to the GPIO pins
+        /// </summary>
+        /// <param name="delaySeconds">The delay in seconds between 0 and 255</param>
+        public void SetPowerOff(byte delaySeconds)
+        {
+            if (delaySeconds < 0)
```

Comment:
with a byte, that should never happen :-) Now with a TimeSpan neither, but you should check the maximum number of seconds